### PR TITLE
Add static support for snippets

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Taking over the snippet functionality from https://github.com/sam-cogan/arm-snippets-vscode.  This first version is just a static set of snippets.  For JSON/JSONC, it contains only snippets for creating the skeleton of an ARM template or ARM parameters file.  For arm-template, it contains all the current snippets from https://github.com/sam-cogan/arm-snippets-vscode.

Need a write-up for the readme.

### CODE REVIEW

File: assets/armsnippets.jsonc [hunk 1]
```diff
@@ -0,0 +1,2406 @@
+{ /*
```
Comment (bhsubra) [inline assets/armsnippets.jsonc]: Can we fix the formatting here?

File: assets/jsonsnippets.jsonc [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+    /*
```
Comment (bhsubra) [inline assets/jsonsnippets.jsonc]: Can we fix the formatting here?

Comment (StephenWeatherford) [reply assets/jsonsnippets.jsonc]: There are some issues I'm going to need to go through.  I consider this just a initial checkin.

Comment (StephenWeatherford) [reply assets/jsonsnippets.jsonc]: I'd fix them now but I'd rather have an initial checkin that has exactly the previous snippets so we can track changes.

**Pull request merged**